lib/internal/validators.js

```diff
@@ -70,6 +70,9 @@ function parseFileMode(value, name, def) {
   return value;
 }
 
+/**
+ * @type {(function(unknown, string, number=, number=): void)}
+ */
```

Comment:
Do you even need `@template`? On my IDE the following seems to work just fine:

```suggestion
/**
 * @callback validateInteger
 * @param {unknown} value
 * @param {string} name
 * @param {number} [min]
 * @param {number} [max]
 */

/** @type {validateInteger} */
```

Reply:
Unfortunately this does not work

<img width="505" alt="Screen Shot 2022-08-16 at 11 21 41 AM" src="https://user-images.githubusercontent.com/1935246/184917194-aeb32d2a-455c-4107-9518-68f078e9f9ed.png">

Reply:
Testing out this PR with VSCode.

With this PR: 
<img width="463" alt="image" src="https://user-images.githubusercontent.com/14309773/185059727-7e7cffcd-b6d5-45c5-b828-7e137d435684.png">
<img width="451" alt="image" src="https://user-images.githubusercontent.com/14309773/185059967-013462b2-21b5-47d8-8ce8-33104f2cea7c.png">


Without this PR: 
<img width="460" alt="image" src="https://user-images.githubusercontent.com/14309773/185060096-70b05b4b-3fed-49cf-a056-02e35bb0ff6d.png">
<img width="449" alt="image" src="https://user-images.githubusercontent.com/14309773/185060216-3ea9c9e2-7702-4b23-bd57-95103d38d0ff.png">

There's definitely an improvement, however it looks like it's not able to pick up the correct types. According to https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html, `@function` is not a supported tag by TypeScript (which I think is what VSCode is using); if I try my suggestion (using `@callback`) here's how it looks like: 
<img width="455" alt="image" src="https://user-images.githubusercontent.com/14309773/185061169-4e8605bb-03c2-4faf-a1a5-926580f2ebc9.png">

Ideally we would find a solution that works everywhere, could you try to find another solution using only the tags supported by TypeScript?

Reply:
I haven't tested this with VSCode, I'll do it. Thanks!

Reply:
I've changed the JSDoc from function to callback. It seems to fix the VSCode, but failed the Webstorm experience. 

I'll try to improve the Webstorm experience as well, but afaik VSCode is used more.

VScode:

<img width="625" alt="Screen Shot 2022-08-17 at 8 28 47 AM" src="https://user-images.githubusercontent.com/1935246/185122534-44ff076a-d1bc-4b02-b9fa-801c6cd124c2.png">

Webstorm:

<img width="563" alt="Screen Shot 2022-08-17 at 8 28 35 AM" src="https://user-images.githubusercontent.com/1935246/185122553-1a12d9c5-3f53-4515-b61b-03e637e23926.png">

Reply:
Both of these editors lack the type checking for the second parameter (if I pass number instead of string).

<img width="554" alt="Screen Shot 2022-08-17 at 8 42 12 AM" src="https://user-images.githubusercontent.com/1935246/185129475-5861f2ee-070f-42ca-9809-e07b28003d94.png">

But Webstorm warns if I change type to function (not callback)

<img width="866" alt="Screen Shot 2022-08-17 at 8 42 50 AM" src="https://user-images.githubusercontent.com/1935246/185129872-e39a7326-2b5a-41a3-9d56-13e9a9214ce8.png">

On callback, webstorm just does not get types at all.

<img width="516" alt="Screen Shot 2022-08-17 at 8 43 22 AM" src="https://user-images.githubusercontent.com/1935246/185130194-943efffb-2b73-4d9e-bd66-8bd5b6d930c6.png">

Reply:
It did not make a difference in terms of type-related warnings in both editors. I think the current state is ready for review.

Reply:
It makes a big difference to me! 

No `@type`: 
<img width="706" alt="image" src="https://user-images.githubusercontent.com/14309773/185147480-1f9c7e86-fa10-4b55-9a97-a905d6111035.png">

With `@type`: 
<img width="758" alt="image" src="https://user-images.githubusercontent.com/14309773/185147576-a907cad3-2cfe-4942-99a4-96f8c6c5527f.png">

Reply:
It seems there has been a miscommunication. My bad! This change is already in the pull request: https://github.com/nodejs/node/pull/44181/files#diff-84047e9de2de28fd3ae796466cebdbae912d724504cb1b44c4f45b3517c1fa5fR92